### DISCUSSION TITLE
Similarity searching

### DISCUSSION THREAD

**Discussion by boglet** [epam/Indigo#1081] [Q&A]
The similarity search is very fast but I can't find a way of ordering hits by similarity score.  Is there a way?  If not, perhaps there should be as users often want to order their hits with the most similar at the top of the list.

**Comment by MysterionRise** [comment]
@boglet which approach you’re using? Are you using plain Indigo or any Bingo flavours?

**Reply by boglet** [reply]
I'm using Bingo (1.9.1.r0-g353401f win64). I can, of course, calculate the similarity for each of the hits and then order my list by similarity but this is relatively slow (using the Indigo toolkit).  What would be useful would be to have something like:

`SELECT regno, smiles, bingo.similarity(smiles,'c1ccccc1') AS sim FROM mytable WHERE smiles @ (0.8,1.0,'c1ccccc1','Tanimoto')::bingo.sim ORDER BY sim DESC;`

This is the Chemaxon equivalent:
`SELECT mol, mol |~| 'CCC' FROM moltable WHERE ('CCC', 0.2)::sim_order |~>| mol;`

This is what I have to do currently in my test harness (after the similarity search) (C#)
```
                IndigoObject queryMol = indigo.loadMolecule(textBox1.Text);
                foreach (DataRow row in ds.Tables[0].Rows)
                {
                    IndigoObject hitMol = indigo.loadMolecule(row.ItemArray[2].ToString());
                    float simval = indigo.similarity(queryMol, hitMol, "tanimoto");
                    dgv1.Rows.Add(row.ItemArray[0], row.ItemArray[1], row.ItemArray[2], simval);
                }
                dgv1.Sort(dgv1.Columns[3], ListSortDirection.Descending);
```
